ludwig/schema/model_config.py

```diff
@@ -512,7 +518,20 @@ def _set_hyperopt_defaults(self):
         # This fills in missing splits, executor config, search_alg, etc.
         self.hyperopt = HyperoptConfig.from_dict(self.hyperopt).to_dict()
 
-        scheduler = self.hyperopt.get("executor", {}).get("scheduler")
+        # Set default num_samples based on search space if not set by user
+        if self.hyperopt.get(EXECUTOR).get(NUM_SAMPLES) is None:
```

Comment:
Is `EXECUTOR` guaranteed to be populated at this point? If so, we can do:

```
if self.hyperopt[EXECUTOR].get(NUM_SAMPLES) is None:
```

Reply:
Yup, it is guaranteed by this point. Will update